designer/app.py

```diff
@@ -760,12 +761,8 @@ def _perform_new(self, *args):
 
         self.close_popup()
 
-        new_proj_dir = os.path.join(get_config_dir(),
-                                    constants.NEW_PROJECT_DIR_NAME)
-        if os.path.exists(new_proj_dir):
-            shutil.rmtree(new_proj_dir)
-
-        os.mkdir(new_proj_dir)
+        new_proj_dir = tempfile.mkdtemp(prefix='designer_',
+                                        dir=get_config_dir())
```

Comment:
If dir is not specified a default directory is used. The default directory is chosen from a platform-dependent list. 

1. On Windows, the directories C:\TEMP, C:\TMP, \TEMP, and \TMP, in that order.
2. On all other platforms, the directories /tmp, /var/folders, and /usr/tmp, in that order.

By default, files in this location may be cleaned (removed) by the system if they are not accessed in 3 days and if not cleaned will not affect designer in any case but give user a chance to save data, if designer is closed due to unwanted reason.

So, 2 option will be better in that case.